docs/source/library-user-guide/building-logical-plans.md

```diff
@@ -19,4 +19,135 @@
 
 # Building Logical Plans
 
-Coming Soon
+A logical plan is a structured representation of a database query that describes the high-level operations and
+transformations needed to retrieve data from a database or data source. It abstracts away specific implementation
+details and focuses on the logical flow of the query, including operations like filtering, sorting, and joining tables.
+
+This logical plan serves as an intermediate step before generating an optimized physical execution plan.
```

Comment:
It might make sense to point them at the architecture guide here for more detail: https://docs.rs/datafusion/latest/datafusion/index.html#query-planning-and-execution-overview